wrangler-service/src/main/java/io/cdap/wrangler/service/directive/DirectivesHandler.java

```diff
@@ -625,7 +625,14 @@ public void execute(HttpServiceRequest request, HttpServiceResponder responder,
             Object object = field.getSecond();
 
             if (object != null) {
-              types.put(fieldName, object.getClass().getSimpleName());
+              Schema schema = convertor.getSchema(object, fieldName);
+              String type = object.getClass().getSimpleName();
+              if (schema != null) {
+                schema = schema.isNullable() ? schema.getNonNullable() : schema;
+                // here we only change the type if this is a logical type to preserve backward compatibility
+                type = schema.getLogicalType() == null ? type : schema.getLogicalType().name();
```

Comment:
better to use schema.getType().name() instead of the classname. Arrays, maps, and nested records will work better that way.

Reply:
would be good to check with @bdmogal on the casing here. This would give it in all caps, which might not be desired.

Reply:
I made the change so the first character is upper case, the rest is lower case